site/src/api/queries/workspaces.ts

```diff
@@ -266,7 +266,12 @@ export const startWorkspace = (
 export const cancelBuild = (workspace: Workspace, queryClient: QueryClient) => {
 	return {
 		mutationFn: () => {
-			return API.cancelWorkspaceBuild(workspace.latest_build.id);
+			const { status } = workspace.latest_build;
+			const params =
+				status === "pending" || status === "running"
+					? { expect_status: status }
+					: undefined;
+			return API.cancelWorkspaceBuild(workspace.latest_build.id, params);
```

Comment:
Double checking... can `cancelWorkspaceBuild` being called in any status? I see the params can be `undefined`, but it would still call the `cancelWorkspaceBuild` 🤔